native/cocos/renderer/pipeline/Define.h

```diff
@@ -330,14 +333,15 @@ struct CC_DLL UBOSkinningAnimation {
     static const String                          NAME;
 };
 
-struct CC_DLL UBOSkinning {
-    static constexpr uint                        JOINTS_OFFSET = 0;
-    static constexpr uint                        COUNT         = UBOSkinning::JOINTS_OFFSET + JOINT_UNIFORM_CAPACITY * 12;
-    static constexpr uint                        SIZE          = UBOSkinning::COUNT * 4;
-    static constexpr uint                        BINDING       = static_cast<uint>(ModelLocalBindings::UBO_SKINNING_TEXTURE);
+class CC_DLL UBOSkinning {
+public:
+    static uint                                  count;
+    static uint                                  size;
+    static constexpr uint                        BINDING            = static_cast<uint>(ModelLocalBindings::UBO_SKINNING_TEXTURE);
     static const gfx::DescriptorSetLayoutBinding DESCRIPTOR;
-    static const gfx::UniformBlock               LAYOUT;
+    static gfx::UniformBlock                     layout;
```

Comment:
Does `gfx::UniformBlock layout` change over time? It is not `const` any more.

Reply:
> Does `gfx::UniformBlock layout` change over time? It is not `const` any more.

 do you mean just changing LAYOUT‘s member variable is a better way?

Reply:
After engine initialized, if user called some related API, will the content change in `layout`?

Reply:
yes. now in engine it's like this. or i should make sure InitLayout just be called once.
```
    public static InitLayout (capacity: number) {
        UBOSkinning._JOINT_UNIFORM_CAPACITY = capacity;
        UBOSkinning._COUNT = capacity * 12;
        UBOSkinning._SIZE = UBOSkinning._COUNT * 4;
        UBOSkinning._LAYOUT = new UniformBlock(SetIndex.LOCAL, UBOSkinning.BINDING, UBOSkinning.NAME, [
            new Uniform('cc_joints', Type.FLOAT4, capacity * 3),
        ], 1);
    }

/**
 * @internal This method only used in Game._initDevice to init localDescriptorSetLayout.layouts[UBOSkinning.NAME]
*/
export function localDescriptorSetLayout_ResizeMaxJoints (maxCount: number) {
    UBOSkinning.InitLayout(maxCount);
    localDescriptorSetLayout.layouts[UBOSkinning.NAME] = UBOSkinning.LAYOUT;
    localDescriptorSetLayout.bindings[UBOSkinning.BINDING] = UBOSkinning.DESCRIPTOR;
}
```